lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -1334,6 +1334,10 @@ bool TypeLoweringVisitor::visitExpr(mlir::UnrealizedConversionCastOp op) {
     return builder->create<mlir::UnrealizedConversionCastOp>(field.type, input)
         .getResult(0);
   };
+  // If the input to the cast is an unknown type, getSubWhatever cannot handle
+  // it, donot lower the op.
+  if (!isa<FVectorType, BundleType, RefType>(op->getOperand(0).getType()))
```

Comment:
I think it may be better to continue to error if we don't support lowering the thing if it's a FIRRTL type?
(only ignore if it's not a FIRRTL type, I mean)

Also I think this maybe should be using `type_isa`, e.g., `!type_isa<FIRRTLType>(...)`?

Reply:
Thanks @dtzSiFive , makes sense, this is required only for other dialects. Updated the check.